pkg/api/server/metrics.go

```diff
@@ -1,24 +1,27 @@
 package server
 
 import (
-	"github.com/99designs/basicauth-go"
 	"net/http"
 
-	"github.com/766b/chi-prometheus"
+	"github.com/99designs/basicauth-go"
+
 	"github.com/go-chi/chi/v5"
+	"github.com/ppaanngggg/chi-prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
 // MetricsMiddleware is the request logger that provides metrics to prometheus
 func metricsMiddleware(name string) func(http.Handler) http.Handler {
-	return chiprometheus.NewMiddleware(name)
+	return chiprometheus.NewPatternMiddleware(name)
 }
 
 func metricsMount(r chi.Router, username, password string) {
 	r.Route("/metrics", func(r chi.Router) {
-		r.Use(basicauth.New("Metrics", map[string][]string{
-			username: {password},
-		}))
+		if username != "" && password != "" {
```

Comment:
You said in the PR description that you are making password optional?
How would this work if you want to use username but password is empty?

Reply:
I want to enable basicauth only when `username` and `password` set

Reply:
TBH I have no knowledge about Prometheus' auth constraints, what is required or possible. 
But your comment about making password optional and the code are incompatible :)